### PR TITLE
Implement LWG-3612: Inconsistent pointer alignment in `std::format`

### LINKED ISSUE #2553: LWG-3612 Inconsistent pointer alignment in `std::format`
LWG-3612 Inconsistent pointer alignment in `std::format`

### PR DESCRIPTION
**Pull request by fsb4000**
Fixes #2553

### CODE REVIEW

File: tests/std/tests/P0645R10_text_formatting_formatting/test.cpp [hunk 1]
```diff
@@ -1322,8 +1322,7 @@ void libfmt_formatter_test_runtime_width() {
     assert(format(STR("{0:{1}}"), 42ull, 7) == STR("     42"));
     assert(format(STR("{0:{1}}"), -1.23, 8) == STR("   -1.23"));
     assert(format(STR("{0:{1}}"), -1.23l, 9) == STR("    -1.23"));
-    assert(format(STR("{0:{1}}"), reinterpret_cast<void*>(0xcafe), 10)
-           == STR("0xcafe    ")); // behavior differs from libfmt, but conforms
```
Comment (AlexGuteniev) [inline tests/std/tests/P0645R10_text_formatting_formatting/test.cpp:1326]: Can you comment on libfmt behavior?

Comment (fsb4000) [reply tests/std/tests/P0645R10_text_formatting_formatting/test.cpp:1326]: Yes, libfmt aligns right: https://gcc.godbolt.org/z/xx466r8dd

**Review by AlexGuteniev — APPROVED** [review]

**Review by StephanTLavavej — APPROVED** [review]

**Review by CaseyCarter — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
I'm mirroring this to the MSVC-internal repo - please notify me if any further changes are pushed.

**Comment by StephanTLavavej** [comment]
Thanks for implementing this LWG issue resolution and improving runtime behavior! :tada: :heart_eyes_cat: :white_check_mark:

**Pull request merged**